mws/mws.py

```diff
@@ -76,18 +85,31 @@ def calc_request_description(params):
     """
     description_items = []
     for item in sorted(params.keys()):
-        encoded_val = quote(str(params[item]), safe='-_.~')
+        encoded_val = params[item]
         description_items.append('{}={}'.format(item, encoded_val))
     return '&'.join(description_items)
 
 
 def remove_empty(dict_):
-    """
-    Returns dict_ with all empty values removed.
-    """
     return {k: v for k, v in dict_.items() if v}
 
 
+def clean_extra_data(extra_data):
+    extra_data_enc = dict()
+    for key, value in extra_data.items():
+        if isinstance(value, (datetime.datetime, datetime.date)):
+            value = str(value.isoformat())
+        if isinstance(value, (bool, int)):
```

Comment:
I think we only really need to check for `bool` here, since we'd like to form the boolean param to its lowercase equivalent (that looks like what MWS expects for requests). I would suggest:

```python
if isinstance(value, bool):
    value = str(value).lower()
```

The other portion for checking an `int` is mostly irrelevant. A later code line, ideally, would just call `str` over all values, which would naturally handle this case.

---

Aside from this, I think we should add a condition at the top of the method that checks for specific data types we know won't work, like `dict`, `list`, `set`, and `tuple`. If we raise an exception for those types, we can clear up a lot of potential confusion early on.